.teamcity/generated/ImageValidation.kts

```diff
@@ -6,116 +6,90 @@ package generated
-//                                        "%docker.deployRepository%teamcity-server:2022.10-nanoserver-1809",
-                                        )
-
-        steps {
-            images.forEach {
-                kotlinFile {
-                    name = "Image Verification - $it"
-
-                    path = "tool/automation/ImageValidation.main.kts"
-                    arguments = "$it"
-                    executionMode = BuildStep.ExecutionMode.ALWAYS
-
-                }
-            }
-        }
-
-
-        failureConditions {
-
-            // fail in case statistics for any image changes for more than N percent
-//            images.forEach {
-//                failOnMetricChange {
-//                    // -- target metric
-//                    param("metricKey", it.replace("%docker.deployRepository%", "").replace("2022.10-", ""))
-//                    units = BuildFailureOnMetric.MetricUnit.PERCENTS
-//                    // -- 5% increase
-//                    threshold = 5
-//                    comparison = BuildFailureOnMetric.MetricComparison.MORE
-//                    compareTo = build {
-//                        buildRule = lastSuccessful()
-//                    }
-//                }
-//            }
-
-
-            failOnText {
-                conditionType = BuildFailureOnText.ConditionType.CONTAINS
-                pattern = "DockerImageValidationException"
-                failureMessage = "Docker Image validation have failed"
-                // allows the steps to continue running even in case of one problem
-                reportOnlyFirstMatch = false
-            }
-        }
-        triggers {
-            finishBuildTrigger {
-                buildType = "${PublishHubVersion.publish_hub_version.id}"
-            }
-        }
-
-        requirements {
-            // -- compatibility with Windows images
-            contains("teamcity.agent.jvm.os.name", "Windows")
-        }
-
-        features {
-            dockerSupport {
-                cleanupPushedImages = true
-                loginToRegistry = on {
-                    dockerRegistryId = "PROJECT_EXT_774,PROJECT_EXT_315"
-                }
-            }
-        }
-//	dependencies {
-//		 dependency(AbsoluteId("TC_Trunk_DockerImages_push_hub_windows")) {
-//			 snapshot { onDependencyFailure = FailureAction.ADD_PROBLEM }
-//		 }
-//		 dependency(AbsoluteId("TC_Trunk_DockerImages_push_hub_linux")) {
-//			 snapshot { onDependencyFailure = FailureAction.ADD_PROBLEM }
-//		 }
-
-        // -- build number dependency
-//        dependency(AbsoluteId("TC_Trunk_BuildDistDocker")) {
-//            snapshot {
-//                reuseBuilds = ReuseBuilds.ANY
-//                onDependencyFailure = FailureAction.IGNORE
-//            }
-//        }
-//	}
-    })
+object image_validation: BuildType({
+
+	name = "Validation of Size Regression - Staging Docker Images (Windows / Linux)"
+	buildNumberPattern="test-%build.counter%"
```

Comment:
Let's remove test- prefix? I'd suggest using the same pattern as was used for the push configurations (buildNumber-counter)

Reply:
@burnasheva yes, good point, I've changed build name pattern in https://github.com/JetBrains/teamcity-docker-images/pull/46